docs/user_guide/preemption.md

```diff
@@ -0,0 +1,249 @@
+| `2-no-guaranteed.queue-2` | 2                                 | 6                                |
+
+![preemption_no_guaranteed_case](../assets/preemption_no_guaranteed_case.png)
+
+### Priority
+
+In general, a high-priority pod can preempt a low-priority pod. You can set the priority by defining a [PriorityClass](https://kubernetes.io/docs/concepts/scheduling-eviction/pod-priority-preemption/) or by utilizing [queue priorities](priorities).
+
+While preemption allows service-type pods to scale up or down through preemption, it can also lead to the preemption of pods that should not be preempted in certain scenarios:
+
+1. Spark Jobs, where the driver pod manages a large number of jobs, and if preempted, all jobs will be affected.
+2. Interactive pods, such as Python notebooks, have a significant impact when restarted and should be avoided from preemption.
+
+To address this issue, we have designed a "do not preempt me" flag. You can set the annotation `yunikorn.apache.org/allow-preemption` to `false` in the PriorityClass to prevent pod requests from being preempted.
+
+### PriorityClass
+
+In this example, we will demonstrate the configuration of `yunikorn.apache.org/allow-preemption` using PriorityClass and observe its effect. The default value for this configuration is set to `true`.
+
+```bash
+apiVersion: scheduling.k8s.io/v1
+kind: PriorityClass
+metadata:
+  name: preemption-not-allow
+  annotations:
+    "yunikorn.apache.org/allow-preemption": "false"
+value: 0
+```
+
+We will deploy 8 pods with a resource requirement of 1 to `queue-1`, `queue-2`, and `queue-3`, respectively. We will deploy to `queue-1` and `queue-2` first, followed by a few seconds delay before deploying to `queue-3`. This ensures that the resource usage in `queue-1` and `queue-2` will be greater than that in `queue-3`, depleting all resources in the parent queue and triggering preemption.
+
+| queue                    | Max Resource | Guaranteed Resource | Types of priorities for Pods |
+| ------------------------ | ------------ | ------------------- | ---------------------------- |
+| `priority-class`         | 16           | x                   |                              |
+| `priority-class.queue-1` | 8            | 3                   | allow-preemption             |
+| `priority-class.queue-2` | 8            | 3                   | preemption-not-allow         |
+| `priority-class.queue-3` | 8            | 3                   | allow-preemption             |
+
+Result:
+
+When preemption is triggered, `queue-3` will start searching for a victim. However, since `queue-2` is set with allow-preemption as false, the resources of `queue-1` will be preempted.
+
+It is important to note that setting `yunikorn.apache.org/allow-preemption` is a strong recommendation but not a guarantee. For instance, DaemonSet can still preempt pods even if allow-preemption is set to false.
```

Comment:
As above, the key point about allow-preemption is that it moves the Pod to the back of the preemption list, but does not guarantee lack of preemption. It affects only the sorting of potential preempted Pods.